src/main/java/org/apache/maven/shared/dependency/analyzer/asm/ConstantPoolParser.java

```diff
@@ -208,7 +208,7 @@ private static String decodeString(ByteBuffer buf) {
         // Explicit cast for compatibility with covariant return type on JDK 9's ByteBuffer
```

Comment:
What about this comment ...?
Look like special cast is needed here.

Reply:
Good point. If this doesn;t fail (and it doesn't) we're missing a test or the comment is no longer true. Hmm, I think that comment applies to line 210 which uses the return type rather than 211 which doesn't but let me double check.